app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/login/LoginViewModel.kt

```diff
@@ -54,32 +54,33 @@ class LoginViewModel @Inject constructor(
     fun verifyToken(apiRequest: ApiTokenRequest) {
         _apiResponse.value = Resource.Loading
         viewModelScope.launch {
-            val response = verifyTokenUseCase(request = apiRequest)
-            _apiResponse.value = response
-            if (response is Resource.Success) {
-                _messageBarState.value = MessageBarState.Success(
-                    message = "Successfully Signed In",
-                )
-            } else if (response is Resource.Error) {
-                _messageBarState.value = MessageBarState.Failure(
-                    message = when (response.ex) {
-                        is SocketTimeoutException -> {
-                            "Connection Timeout Exception"
-                        }
+            verifyTokenUseCase.invoke(request = apiRequest).collect { response ->
+                _apiResponse.value = response
```

Comment:
If we are using a Flow where we emit `Idle` and `Loading`, I think it would be slightly better to drive everything from the Flow. That means we would handle `Resource.Idle` and `Resource.Loading` in the `collect { ` block

Reply:
I've added these to the collect block. In the case of Idle, there is action required. As for Loading, this will be handled in the LoginScreen as it's used to display a progress indicator